### ISSUE TITLE
[package] boost/1.72.0: Addition of BOOST_FILESYSTEM_VERSION breaks previously working builds

### ISSUE THREAD

**Issue by Narwhalrus** [conan-io/conan-center-index#12351]
### Package and Environment Details

* Package Name/Version: **boost/1.72.0**
* Operating System+version: **CentOS7**
* Compiler+version: **GCC 7.3.1**
* Conan version: **conan 1.51.3**
* Python version: **Python 3.6.8**

### Conan profile

Configuration for profile default:

[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=7
compiler.libcxx=libstdc++
build_type=Release
[options]
[conf]
[build_requires]
[env]

### Steps to reproduce

Using conan.cmake

```cmake
...
conan_cmake_configure(REQUIRES boost/1.72.0 GENERATORS cmake_find_package_multi cmake_paths)
conan_cmake_autodetect(settings)
conan_cmake_install(PATH_OR_REFERENCE . BUILD missing REMOTE conancenter SETTINGS ${settings}
include(${CMAKE_CURRENT_BINARY_DIR}/conan_paths.cmake)
...
```

Standard cmake generation

```
$ make
...
/user/.conan/data/boost/1.72.0/_/_/package/56e85e048a237371c86d912df0dad3ef833e6dcd/include/boost/filesystem/config.hpp:16:5: error: #error Compiling Filesystem version 3 file with BOOST_FILESYSTEM_VERSION defined != 3
 #   error Compiling Filesystem version 3 file with BOOST_FILESYSTEM_VERSION defined != 3
     ^~~~~
```

### Logs

<details><summary>Click to expand log</summary>

```
Put your log output here
```

</details>

**Comment by garethsb** [comment]
Should be fixed by #12321

**Comment by Narwhalrus** [comment]
Thank you for the quick response.

In the meantime, I've made the following change to my CMakeLists.txt.

```cmake
...
conan_cmake_configure(REQUIRES 
					  boost/1.72.0
                      		       GENERATORS
                        		  cmake_find_package_multi
                        		  cmake_paths
                                       OPTIONS
                                         boost:filesystem_version=3) <-- this
...
```

**Comment by bryanjhunt** [comment]
Not sure what happened here, but this broke our builds yesterday too. Please when adding new options, make the default be equal to the previous behavior. Thanks for fixing this so quickly though.

**Comment by uilianries** [comment]
Fixed by #12321 Please, update your local Boost copy and try again. In case that error persists, re-open this issue and comment with more details.
